FileSystem/FileMapperMocker.js

```diff
@@ -10,28 +25,490 @@ var USE_FILEMAPPER_MOCKER = true;
+ * @todo Set FileMapper Mocker MimeType by file extension
+ * @body Infer MimeType from file extension. If no extension is provided, assume
+ * PLAIN_TEXT
+ * @param {string} localPath Local file path of the file
+ * @return {string} File Id of the newly created file
+ */
+function createFile2(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      root = root.createFolder(pathSplit[i]);
+    } else {
+      root = folderIterator.next();
+    }
+  }
+  var fileName = pathSplit[pathSplit.length - 1];
+  // Create file and return file id
+  return root.createFile(fileName, MimeType.PLAIN_TEXT).getId();
+}
+
+/**
+ * Create folder API
+ * The API creates intermediate folders if required.
+ * @param {string} localPath Local file path of the folder
+ * @return {string} Folder Id of the newly created folder
+ */
+function createFolder(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  for (var i = 0; i < pathSplit.length; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      root = root.createFolder(pathSplit[i]);
+    } else {
+      root = folderIterator.next();
+    }
+  }
+  return root.getId();
+}
+
+/**
+ * Move file API
+ * API moves all files to the target folder. Apps Script does
+ * not have a move API. Move is implemented by making a copy
+ * and deleting the original file.
+ * @param {Array} sourceFilePaths List of local files to be moved
+ * @param {string} targetFolderPath Destination folder path
+ */
+function moveFile(sourceFilePaths, targetFolderPath) {
+  var targetFolderId;
+  // Get folder, if it doesn't exist then create it
+  try {
+    targetFolderId = this.getFolderId(targetFolderPath);
+  } catch (e) {
+    targetFolderId = this.createFolder(targetFolderPath);
+  }
+  var targetFolder = DriveApp.getFolderById(targetFolderId);
+  // Move all files
+  for (var i = 0; i < sourceFilePaths.length; i++) {
+    var fileId = this.getFileId(sourceFilePaths[i]);
+    var file = DriveApp.getFileById(fileId);
+    file.makeCopy(file.getName(), targetFolder);
+    file.setTrashed(true);
+  }
+}
+
+/**
+ * Move folder API
+ * API moves all folders to the target folder. Apps Script does
+ * not have a move API. Move is implemented by making a copy
+ * and deleting the original folder. Apps Script does not have a
+ * direct copy folder API as well, a copy is done recursively
+ * @param {Array} sourceFilePaths List of local folders to be moved
+ * @param {string} targetFolderPath Destination folder path
+ */
+function moveFolder(sourceFilePaths, targetFolderPath) {
```

Comment:
Code for moveFile/moveFolder looks similar. Create a helper method for similar operations?

Reply:
Fixed: Made helper function cloneFileUtil for moveFile and copyFile. Similarly, made cloneFolderUtil for copyFolder and moveFolder